tests/docs/pytest.voq_chassis.md

```diff
@@ -0,0 +1,299 @@
+
+
+# About this Manual
+This document describes the design details for testing SONiC Chassis by extending the existing sonic-mgmt test framework.  
+
+# Scope
+A SONiC Chassis typically has multiple line cards and supervisor card. Today, there is no 'single point of mgmt' for a SONiC Chassis. In a dis-aggregated model, each linecard
+and supervisor card has its own management IP address, that can be controlled/configured independently. 
+
+The existing sonic-mgmt test framework has well defined test methodologies for testing of SONiC pizza boxes (a single mgmt. IP). 
+The scope of this document is to enhance the framework to test SONiC Chassis with multiple mgmt. IP addresses, while still utilizing the existing test methodologies. 
+
+The end goal is to be able to run the existing OC tests against a SONiC Chassis with minimal changes to test cases itself.
+
+# 1 Requirements
+The sonic-mgmt test framework enhancements for testing on SONiC VOQ shall support the following
+- backward compatability - Any enhancement should not break testing of pizza box, while allowing for testing of SONiC Chassis.
+
+ - Test case changes - The enhancements should allow for incremental changes to existing test cases to support testing of SONiC Chassis and pizza boxes. If a test is enhanced to work with SONiC Chassis, it should still be able to test a SONiC pizza box.
+
+# 2 Approach
+In order to test SONiC chassis, the following issues had to be resolved:
+- The SONiC chassis as a collection of linecards. Each linecard has its own mgmt. IP. 
+- The front-panel linecards could be multi-asic (have multiple SONiC instances). 
+- Need to distinguish between front-panel linecards and supervisor cards
+
+The solution is to improve the current sonic-mgmt test infrastructure to support multi-DUT and multi-ASIC systems. Then, a SONiC chassis can be viewed as a collection of DUT's, each with either a single or multiple asics. This includes the 'supervisor' card as well. 
+
+There was support added for multi-DUT for [Spytest](https://github.com/Azure/sonic-mgmt/pull/1848). This approach was extended with the following enhancements that facilitate the testing of SONiC chassis.
+
+## 2.1 Ansible modules
+The OC pytest tests in sonic-mgmt use ansible modules for querying for parsing data from the DUT. 
+
+The ansible modules that are applicable to be run in different namespaces have been enhanced to support such functionaliy. Currently, the following modules support running them on different namespaces:
+- bgp_facts - added 'instance_id' as an optional module param.
+- config_facts - added 'namespace' as an opitonal module param.
+
+If the optional namespace/ASIC specific module params are not specified, then they default to returning the output from the global namespace.
+
+Other modules will be enhanced as needed during adaption of the tests to SONiC chassis.
+
+## 2.2 Parameterizing dut_index and asic_index 
```

Comment:
The infrastructure has been extended to support below fixtures for parameterizing test cases:
* enum_dut_index
* enum_dut_hostname
* enum_asic_index
* enum_frontend_asic_index
* enum_dut_portname
* enum_dut_portname_oper_up
* enum_dut_portname_admin_up
* enum_dut_portchannel
* enum_dut_portchannel_oper_up
* enum_dut_portchannel_admin_up
* enum_dut_feature
Please update this document accordingly.